### ISSUE TITLE
Cannot run "occ db:convert-filecache-bigint" (after update from NC18 to NC19)

### ISSUE THREAD

**Issue by tchap2** [nextcloud/server#24128]
After 

Running NC, in a docker, on UNRAID OS. After update to NC19 I get:

_Some columns in the database are missing a conversion to big int. Due to the fact that changing column types on big tables could take some time they were not changed automatically. By running 'occ db:convert-filecache-bigint' those pending changes could be applied manually. This operation needs to be made while the instance is offline. For further details read the documentation page about this.

mounts.storage_id
mounts.root_id
mounts.mount_id_

This does not solve the problem: https://docs.nextcloud.com/server/19/admin_manual/configuration_database/bigint_identifiers.html

I can update from terminal only, which runs only if NC is up an running. If I shut down NC, I cannot access terminal.

Can I just change these in mariadb backend?

**Comment by tjareson** [comment]
Same issue here, but with 20.04 (very small installation) in docker. 
I run occ db:convert-filecache-bigint and simply nothing happens. No difference, if I switch the instance into maintenance mode or not.

**Comment by mbarczak** [comment]
Same issue here. List of tables : 
```
federated_reshares.share_id
filecache_extended.fileid
share_external.id
share_external.parent
```
from which only  ```filecache_extended.fileid``` contains about 54k rows. Command has been running for 20hours, without an effect.

**Comment by mbarczak** [comment]
OK, I think I got it. 
#### TLDR : use -t -i , while calling docker exec command :
```bash
docker exec -i -t --user www-data nextcloud_fpm php /var/www/html/occ db:convert-filecache-bigint
# -i is the key here : gives us interactive shell
# -t - allocates a pseudo-TTY, which gives as proper colors
# nextcloud_fpm should be replaced with the name of your nextcloud container
```
#### Explanation:
I've put some debug info before and after call to confirmation prompt, in the file _core/Command/Db/ConvertFilecacheBigInt.php_ : 
```php
		$output->writeln('<comment>This can take up to hours, depending on the number of files in your instance!</comment>');
		if ($input->isInteractive()) {
			$helper = $this->getHelper('question');
			$question = new ConfirmationQuestion('Continue with the conversion (y/n)? [n] ', false);
			$output->writeln('<comment>MB: Before asking for confirmation</comment>');
			if (!$helper->ask($input, $output, $question)) {
				return 1;
			}
			$output->writeln('<comment>MB: After asking for confirmation</comment>');
		}
		$this->connection->migrateToSchema($schema->getWrappedSchema());
		return 0;
	}
}
```
and I was calling my original conversion command like this : 
```bash
docker exec --user www-data nextcloud_fpm php /var/www/html/occ db:convert-filecache-bigint
```
Than following was happening : 

1. First debug (_Before_) has been shown before the confirmation prompt,
2. The confirmation prompt was asking for pressing 'y',
3. After pressing 'y', the second debug (_After_) was not shown.

Which gives a hint, that the command is not hanging because of issue with _db:convert-filecache-bigint_ implementation, but it's not receiving the confirmation key.  A quick google for _docker exec read from stdin_ gave the answer :)

**Comment by tjareson** [comment]
@mbarczak thanks for debugging it. That was fixing it. Problem solved. Thanks! :-)

**Comment by anoxi** [comment]
@mbarczak thanks, your solution works for me, too.

**Comment by szaimen** [comment]
Seems like you'll found a solution. Thanks for sharing!
